exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsConfigHelper.java

```diff
@@ -5,79 +5,76 @@
 
 package io.opentelemetry.exporter.internal;
 
-import com.google.common.annotations.VisibleForTesting;
-import java.util.logging.Logger;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
 import javax.annotation.Nullable;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
 import javax.net.ssl.SSLException;
-import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
 import javax.net.ssl.X509KeyManager;
 import javax.net.ssl.X509TrustManager;
 
 /**
- * Utility class to help with management of TLS related components. This class is ultimately
- * responsible for enabling TLS via callbacks passed to the configure[...]() methods. This class is
- * only intended for internal OpenTelemetry exporter usage and should not be used by end-users.
+ * Utility class to help with management of TLS related components. TLS config consists {@link
+ * #keyManager}, {@link #trustManager}, which combine to form {@link #sslContext}. These components
+ * can be configured via higher level APIs ({@link #createTrustManager(byte[])} and {@link
+ * #createKeyManager(byte[], byte[])}) which parse keys in PEM format, or the lower level API {@link
+ * #setSslContext(SSLContext, X509TrustManager)} in which the components are directly set, but NOT
+ * both. Attempts to reconfigure components which have already been configured throw {@link
+ * IllegalStateException}. Consumers access components via any combination of {@link
+ * #getKeyManager()}, {@link #getTrustManager()}, and {@link #getSslContext()}.
```

Comment:
Simplified this class a bit now that the contract is simpler. This javadoc is a good explanation of the behavior.